Opensearch

MichaelLangbein opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The Project is [building and passes all tests](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/DEVELOPMENT.md#further-you-can-test-and-build-locally) without errors


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
This merge-request summarizes the work done during the flood-system sprint.

Issue Number: N/A


## What is the new behavior?
Two new clients have been added to `services-ogc`:
 - `CswClient`: a javascript-client implementing the [CSW specification](https://www.ogc.org/standards/cat)
 - `OpensearchWrapperService`: a wrapper around [EOXC opensearch-client](https://www.npmjs.com/package/opensearch-browser). It exposes the clients URL-parsing mechanism to allow the end user to search not only `results`-links, but also `collection` links.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
All changes affect only `services-ogc`.

boeckMt commented on projects/services-ogc/src/lib/opensearch/README.md:
```diff
@@ -0,0 +1,6 @@
+## Native support
+
+Most browser have a method `window.external.AddSearchProvider` that speaks opensearch.
```
Window.external is **Deprecated** - https://developer.mozilla.org/en-US/docs/Web/API/Window/external

MichaelLangbein replied:
Thanks for the link! Removed.

boeckMt commented on projects/services-ogc/tsconfig.spec.json:
```diff
@@ -5,7 +5,7 @@
       "jasmine",
       "node"
-    ]
+    ],
```
Do not use trailing commas in JSON.

MichaelLangbein replied:
done

boeckMt commented on package.json:
```diff
@@ -98,6 +99,7 @@
     "karma-jasmine": "~3.3.0",
     "karma-jasmine-html-reporter": "^1.5.0",
     "karma-junit-reporter": "^1.2.0",
+    "license-checker": "^25.0.1",
```
Why did you add license-checker, there are no changes in the demo for license or an automated use at buildtime.

MichaelLangbein replied:
That's indeed superfluous. Removed! Thanks!

boeckMt commented:
@MichaelLangbein i added some comments to your PR.

On the next time please try to stick to our [Commit Message Guidelines](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/CONTRIBUTING.md#commit) so in case we want to use a automated Changelog/Release in some time it can be created from the keywords.

MichaelLangbein commented:
> @MichaelLangbein i added some comments to your PR.

Have I missed any of those comments? I've addressed all comments visible in this conversation. Anything else left to do? I'm more than happy to add any required additional changes!

MichaelLangbein commented:
Awesome, thanks!
